doc/architecture.md

```diff
@@ -0,0 +1,30 @@
+# Architecture
+
+This document describes the high-level architecture of plrust.
+If you want to familiarize yourself with the code base, you are just in the right place!
+
+
+## Bird's Eye View
+
+![](assets/architecture_1.png)
+
+
```

Comment:
Maybe this is neither here nor there, but I am wondering how far we could get with the new [mermaid](https://github.blog/2022-02-14-include-diagrams-markdown-files-mermaid/) syntax instead of image files? 

On one hand, it'd be cool to not have binary file revisions in the git history every time the architecture needed to change, but on the other hand it'd be another syntax to learn. I am torn.